src/vivarium/framework/components/parser.py

```diff
@@ -128,60 +154,77 @@ def parse_component_config(
 
         Parameters
         ----------
-        component_config
-            A hierarchical component specification blob.
+        level
+            A level of the hierarchical component specification blob.
+        prefix
+            A list of strings representing the import path prefix.
 
         Returns
         -------
-            A flat list of strings, each string representing the full python import
-            path to the component, the component name, and any arguments.
+        List[Component]
+            A flat list of Components
         """
-        if not component_config:
-            return []
-
-        return self._process_level(component_config, [])
-
-    def _process_level(self, level, prefix):
         if not level:
             raise ParsingError(
-                f"Check your configuration. Component {prefix} should not be left empty with the header"
+                f"Check your configuration. Component {prefix} should not "
+                "be left empty with the header"
             )
 
-        if isinstance(level, list):
-            return [".".join(prefix + [child]) for child in level]
-
         component_list = []
-        for name, child in level.items():
-            component_list.extend(self._process_level(child, prefix + [name]))
+        if isinstance(level, dict):
+            for name, child in level.items():
+                component_list.extend(self.process_level(child, prefix + [name]))
+        else:
```

Comment:
Any way an improperly-formatted yaml would result in this not being a list?

Reply:
Yes that is possible (and was possible before as well). I can make a change to catch this case, though I'm not sure how important it is

Reply:
Done this now and added test coverage for it @stevebachmeier